src/main/java/org/opensearch/performanceanalyzer/decisionmaker/deciders/searchbackpressure/SearchBackPressurePolicy.java

```diff
@@ -0,0 +1,317 @@
+        this.searchBackPressureClusterRCA = searchBackPressureClusterRCA;
+        this.searchBackPressureShardHeapIncreaseAlarm = searchBackPressureShardHeapIncreaseAlarm;
+        this.searchBackPressureShardHeapDecreaseAlarm = searchBackPressureShardHeapDecreaseAlarm;
+        this.searchBackPressureTaskHeapIncreaseAlarm = searchBackPressureTaskHeapIncreaseAlarm;
+        this.searchBackPressureTaskHeapDecreaseAlarm = searchBackPressureTaskHeapDecreaseAlarm;
+        LOG.info("SearchBackPressurePolicy#SearchBackPressurePolicy() initialized");
+    }
+
+    public SearchBackPressurePolicy(SearchBackPressureClusterRCA searchBackPressureClusterRCA) {
+        this(searchBackPressureClusterRCA, null, null, null, null);
+    }
+
+    /**
+     * records issues which the policy cares about and discards others
+     *
+     * @param issue an issue with the application
+     */
+    private void record(HotResourceSummary issue) {
+        // TODO: change nested if into menaing methods like recordSearchBpShardjIssue()
+        // recordSearchBpTaskIssue()
+
+        LOG.info("SearchBackPressurePolicy#record()");
+        if (HEAP_SEARCHBP_SHARD_SIGNALS.contains(issue.getResource())) {
+            LOG.info("Recording shard-level issue");
+            recordSearchBackPressureShardIssue(issue);
+        } else if (HEAP_SEARCHBP_TASK_SIGNALS.contains(issue.getResource())) {
+            LOG.info("Recording Task-Level issue");
+            recordSearchBackPressureTaskIssue(issue);
+        }
+    }
+
+    private void recordSearchBackPressureShardIssue(HotResourceSummary issue) {
+        // increase alarm for heap-related threshold (shard-level)
+        if (issue.getMetaData() == SearchBackPressureRcaConfig.INCREASE_THRESHOLD_BY_JVM_STR) {
+            LOG.info("recording increase-level issue for shard");
+            searchBackPressureShardHeapIncreaseAlarm.recordIssue();
+        }
+
+        // decrease alarm for heap-related threshold (shard-level)
+        if (issue.getMetaData() == SearchBackPressureRcaConfig.DECREASE_THRESHOLD_BY_JVM_STR) {
+            LOG.info("recording decrease-level issue for shard");
+            searchBackPressureShardHeapDecreaseAlarm.recordIssue();
+        }
+    }
+
+    private void recordSearchBackPressureTaskIssue(HotResourceSummary issue) {
+        // increase alarm for heap-related threshold (task-level)
+        if (issue.getMetaData() == SearchBackPressureRcaConfig.INCREASE_THRESHOLD_BY_JVM_STR) {
+            LOG.info("recording increase-level issue for task");
+            searchBackPressureTaskHeapIncreaseAlarm.recordIssue();
+        }
+
+        // decrease alarm for heap-related threshold (task-level)
+        if (issue.getMetaData() == SearchBackPressureRcaConfig.DECREASE_THRESHOLD_BY_JVM_STR) {
+            LOG.info("recording decrease-level issue for task");
+            searchBackPressureTaskHeapDecreaseAlarm.recordIssue();
+        }
+    }
```

Comment:
Can we make this modular and avoid the code duplication? 

`SearchBackPressureShardIssue` and `SearchBackPressureTaskIssue` can implement an interface `SearchBackPressureIssue` .

Reply:
Do you mean a functional interface? 

Or can it be a function named recordSearchBackPressureIssue(HotResourceSummary issue, boolean isShardlevel). if it is in shard level, then it records issue for searchBackPressureShard alarms, otherwise, it records issue for searchBackPressureTask Alarms?

Reply:
Current Solution is to refactor the 2 functions into 1 general action.
private void recordSearchBackPressureIssue(HotResourceSummary issue, boolean isShard)